pandas/core/categorical.py

```diff
@@ -263,18 +259,12 @@ def __init__(self, values, categories=None, ordered=None, name=None, fastpath=Fa
 
         if categories is None:
             try:
-                codes, categories = factorize(values, sort=True)
-                # If the underlying data structure was sortable, and the user doesn't want to
-                # "forget" this order, the categorical also is sorted/ordered
-                if ordered is None:
-                    ordered = True
+                codes, categories = factorize(values, sort=ordered)
             except TypeError:
-                codes, categories = factorize(values, sort=False)
-                if ordered:
-                    # raise, as we don't have a sortable data structure and so the user should
-                    # give us one by specifying categories
-                    raise TypeError("'values' is not ordered, please explicitly specify the "
-                                    "categories order by passing in a categories argument.")
+                # raise, as we don't have a sortable data structure and so the user should
```

Comment:
This is the test which triggered this codepath:
```
    def test_constructor_unsortable(self):

        # it works!
        arr = np.array([1, 2, 3, datetime.now()], dtype='O')
        factor = Categorical.from_array(arr)
        self.assertFalse(factor.ordered)
```

e.g. this should give different results now:
```
        arr = np.array([1, 2, 3, datetime.now()], dtype='O')
        factor = Categorical(arr, ordered=True)
```
before it errored with a type error, now it either succeeds or errors in the factorize step, but it should print the warning/raises a error that the values are not sortable and the user should supply a user-defined order via supplied `categories=[...]`